### PR TITLE
wpt serve should return non-zero if any of the subprocess fails

### PR DESCRIPTION
**Pull request by clopez**
When executing two times `./wpt serve` on the same machine, the second run fails to bind to the port and `wpt serve` fails to start (as expected), but it was exiting with zero status despite of the failure to start.

Fix this by exiting on the main wpt program with zero status only if all the the subprocess exit with zero, otherwise exit with N, being N the number of subprocess that have failed.

//cc @Ms2ger @jgraham

### CODE REVIEW

File: tools/serve/serve.py [hunk 1]
```diff
@@ -922,17 +922,22 @@ def handle_signal(signum, frame):
...
-
-            for item in iter_procs(servers):
-                logger.info("Status of %s:\t%s" % (item.name, "running" if item.is_alive() else "not running"))
+                for subproc in iter_procs(servers):
+                    subproc.join(1)
+
+            failed_subproc = 0
+            for subproc in iter_procs(servers):
+                if subproc.is_alive():
+                    logger.info('Status of subprocess "%s": running' % subproc.name)
```
Comment (Hexcles) [inline tools/serve/serve.py:933]: Shouldn't we kill the subprocess here? Otherwise it'd become a daemon right?

Comment (clopez) [reply tools/serve/serve.py:933]: As far as I can test, they don't become daemons. Not sure now how they get killed, but they do (at least when ending "wpt serve" via Ctrl+C)
Do you think it's better (or more clear) to send a kill signal here?

Comment (Hexcles) [reply tools/serve/serve.py:933]: Ctrl-C in an interactive shell not only sends SIGINT to the parent process, but also to *all* subprocesses in the process group, so effectively each subprocess gets terminated independently. If you call `kill -INT`, you should see the subprocesses become daemons.

We can address this in another PR though. This PR on its own arguably fixes a bug (the exit code shouldn't be 0), whereas the change of signal handling is a nice-to-have improvement.

**Review by Hexcles — APPROVED** [review]

File: tools/serve/serve.py (hunk 1 above)
Comment (clopez) [reply tools/serve/serve.py:933]: Good point about Ctrl-C sending SIGINT to all process from an iterative shell. However, I tried now to run ```kill -INT``` over the main pid of "wpt serve" and I still see all the subprocess ending with the main wpt (no daemons left). I'm a bit puzzled, not sure about what causes them to be closed.

Comment (Hexcles) [reply tools/serve/serve.py:933]: For the record, I've found out how these subprocesses are terminated. They all have `daemon=True`: https://docs.python.org/2/library/multiprocessing.html#multiprocessing.Process.daemon so they will be closed by Python runtime when the main process exits.

**Pull request merged**